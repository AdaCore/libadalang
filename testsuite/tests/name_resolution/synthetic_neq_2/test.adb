procedure Test is
   package Foo is
      type T is null record;

      function "=" (A, B : T) return Boolean is (True);
   end Foo;

   package Bar is
      type U is new Foo.T;
   end Bar;

   use Bar;

   A, B : U;
begin
   if A /= B then
      raise Program_Error;
   end if;
   pragma Test_Statement;

   if "/=" (A, B) then
      raise Program_Error;
   end if;
   pragma Test_Statement;
end Test;
