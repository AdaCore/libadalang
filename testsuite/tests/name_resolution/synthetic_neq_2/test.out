Analyzing test.adb
##################

Resolving xrefs for node <IfStmt test.adb:16:4-18:11>
*****************************************************

Expr: <RelationOp test.adb:16:7-16:13>
  type:          <ConcreteTypeDecl ["Boolean"] __standard:3:3-3:33>
  expected type: <ConcreteTypeDecl ["Boolean"] __standard:3:3-3:33>
Expr: <Id "A" test.adb:16:7-16:8>
  references:    <DefiningName "A" test.adb:14:4-14:5>
  type:          <ConcreteTypeDecl ["U"] test.adb:9:7-9:27>
  expected type: <ConcreteTypeDecl ["U"] test.adb:9:7-9:27>
Expr: <OpNeq "/=" test.adb:16:9-16:11>
  references:    <DefiningName ""="" test.adb:5:16-5:19>
  type:          None
  expected type: None
Expr: <Id "B" test.adb:16:12-16:13>
  references:    <DefiningName "B" test.adb:14:7-14:8>
  type:          <ConcreteTypeDecl ["U"] test.adb:9:7-9:27>
  expected type: <ConcreteTypeDecl ["U"] test.adb:9:7-9:27>

Resolving xrefs for node <IfStmt test.adb:21:4-23:11>
*****************************************************

Expr: <CallExpr test.adb:21:7-21:18>
  references:    <DefiningName ""="" test.adb:5:16-5:19>
  type:          <ConcreteTypeDecl ["Boolean"] __standard:3:3-3:33>
  expected type: <ConcreteTypeDecl ["Boolean"] __standard:3:3-3:33>
Expr: <Str ""/="" test.adb:21:7-21:11>
  references:    <DefiningName ""="" test.adb:5:16-5:19>
  type:          <ConcreteTypeDecl ["Boolean"] __standard:3:3-3:33>
  expected type: None
Expr: <Id "A" test.adb:21:13-21:14>
  references:    <DefiningName "A" test.adb:14:4-14:5>
  type:          <ConcreteTypeDecl ["U"] test.adb:9:7-9:27>
  expected type: <ConcreteTypeDecl ["U"] test.adb:9:7-9:27>
Expr: <Id "B" test.adb:21:16-21:17>
  references:    <DefiningName "B" test.adb:14:7-14:8>
  type:          <ConcreteTypeDecl ["U"] test.adb:9:7-9:27>
  expected type: <ConcreteTypeDecl ["U"] test.adb:9:7-9:27>


Done.
